### PR TITLE
Release Kedro 0.18.4

### LINKED ISSUE #2004: Release Kedro `0.18.4`
### Depends on:
- Dataset issues
- Spaceflights tutorial documentation
- Open PRs related to datasets:
   - [x] https://github.com/kedro-org/kedro/pull/2082
   - [x] https://github.com/kedro-org/kedro/pull/1746
   - [x] https://github.com/kedro-org/kedro/pull/1992
   - [x] https://github.com/kedro-org/kedro/pull/1865
   - [x] https://github.com/kedro-org/kedro/pull/1312
   - [x] https://github.com/kedro-org/kedro/pull/1844
   - [x] https://github.com/kedro-org/kedro/pull/1962
   - [x] https://github.com/kedro-org/kedro/pull/1964
   - [x] https://github.com/kedro-org/kedro/pull/1931
   - [x] https://github.com/kedro-org/kedro/pull/1587

For the above PRs: if it's nearly finished, but the author isn't responding, we as a team can take over and finish the PR. If the PR still needs a lot of work and the author isn't responding, I suggest we close it and ask them to re-open in the new `kedro-datasets` repo.

### PR DESCRIPTION
**Pull request by ankatiyar**
Signed-off-by: Ankita Katiyar <ankitakatiyar2401@gmail.com>

> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Release Kedro 0.18.4 🎉 Will resolve #2004 

## Development notes

- Updated kedro version 
- Added the new team members to `CITATIONS.cff`
## Checklist

- [ ] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -33,9 +39,11 @@ authors:
...
   given-names: Joel
+- family-names: Stichbury
+  given-names: Jo
 - family-names: Theisen
   given-names: Merel
 title: Kedro
-version: 0.18.3
-date-released: 2022-09-20
+version: 0.18.4
+date-released: 2022-12-05
```
Comment (ankatiyar) [inline CITATION.cff:48]: Will update the date according to when it goes out

File: docs/source/deployment/databricks.md [hunk 2]
```diff
@@ -169,10 +169,10 @@ In your newly-created notebook, put each of the below code snippets into a separ
 %sh rm -rf ~/projects/iris-databricks && git clone --single-branch --branch main https://${GITHUB_USER}:${GITHUB_TOKEN}@github.com/${GITHUB_USER}/<your-repo-name>.git ~/projects/iris-databricks
 ```
 
-* Install the latest version of Kedro compatible with version `0.18.3`
+* Install the latest version of Kedro compatible with version `0.18.4`
 
 ```console
-%pip install "kedro[spark.SparkDataSet]~=0.18.3"
+%pip install "kedro[spark.SparkDataSet]~=0.18.4"
```
Comment (merelcht) [inline docs/source/deployment/databricks.md:175]: I just remembered I need to add an issue to update the docs to point to `kedro-datasets[XXX]` after we release `1.0.0`.

**Review by merelcht — APPROVED** [review]
image

**Review by SajidAlamQB — APPROVED** [review]
meme 0 18 4
🚀🚀🚀

**Pull request merged**